File: app/src/main/java/pratamawijaya/moviesinfokotlin/app/AppDependencies.java
package pratamawijaya.moviesinfokotlin.app;

import android.app.Application;
import pratamawijaya.moviesinfokotlin.data.network.ServiceDependencies;
import pratamawijaya.moviesinfokotlin.domain.executor.PostExecutionThread;
import pratamawijaya.moviesinfokotlin.domain.executor.ThreadExecutor;
import pratamawijaya.moviesinfokotlin.presentation.di.UIDependencies;

/**
 * Created by pratama on 9/30/16.
 */

public interface AppDependencies extends ServiceDependencies, UIDependencies {
  Application application();

  ThreadExecutor threadExecutor();

  PostExecutionThread postExecutionThread();
}

File: app/src/main/kotlin/pratamawijaya/moviesinfokotlin/app/AppDependencies.kt
package pratamawijaya.moviesinfokotlin.app

import android.app.Application
import pratamawijaya.moviesinfokotlin.data.network.ServicesDependencies
import pratamawijaya.moviesinfokotlin.domain.executor.PostExecutionThread
import pratamawijaya.moviesinfokotlin.domain.executor.ThreadExecutor

/**
 * Created by Pratama Nur Wijaya
 * Date : Oct - 10/11/16
 * Project Name : MoviesInfoKotlin
 */
interface AppDependencies : ServicesDependencies {
  fun application(): Application

  fun threadExecutor(): ThreadExecutor

  fun postExecutionThread(): PostExecutionThread

}